src/pages/MovieDetails.js

```diff
@@ -1,25 +1,76 @@
 import React, { Component } from 'react';
+import { Link, Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
 
 import * as movieAPI from '../services/movieAPI';
-import { Loading } from '../components';
+import { Loading } from '../components'; // ele vai saber que é o index
 
 class MovieDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movie: {},
+      isLoaded: true,
+      idEdit: '0', // para armazenar o id do movie que vem por url
+      shouldRedirect: false,
+    };
+    this.getMovieAPI = this.getMovieAPI.bind(this);
+    this.OnDeleteMovie = this.OnDeleteMovie.bind(this);
+  }
+
+  async componentDidMount() {
+    this.getMovieAPI();
+  }
+
+  async getMovieAPI() {
+    const { id } = this.props.match.params; // pegando o id da URL
+    const movieResult = await movieAPI.getMovie(id);
+    this.setState({ movie: movieResult, isLoaded: false, idEdit: id });
+  }
+
+  OnDeleteMovie() {
```

Comment:
Ju, acho que por convenção, o méttodos/funções dentro de uma classe começam com letra minúscula
```suggestion
  onDeleteMovie() {
```